BlueDBOnDisk/src/main/java/io/bluedb/disk/segment/Segment.java

```diff
@@ -200,11 +199,8 @@ protected static boolean doesfileNameRangeOverlap(File file, long min, long max
 		for (long rollupLevel: ROLLUP_LEVELS) {
 			Path path = getPathFor(groupingNumber, rollupLevel);
 			BlueReadLock<Path> lock = lockManager.acquireReadLock(path);
-			try {
-				if (lock.getKey().toFile().exists()) {
-					return lock;
-				}
-			} catch (Throwable t) { // make sure we don't hold onto the lock if there's an exception
+			if (lock.getKey().toFile().exists()) {

```

Comment:
Is it not possible to throw an exception here then? Seems scary to not be closing the lock in a finally

Reply:
I think we can add a try-catch block that only closes the lock if we have an exception